From 12df44c10f2593166d3b56c2c1639b498c10edf6 Mon Sep 17 00:00:00 2001
From: Sidraya Jayagond <sidraya.bj@pathparntertech.com>
Date: Wed, 30 Jun 2021 17:01:49 +0530
Subject: [PATCH 27/29] media: platform: Kconfig: Add Video decoder kconfig
 entries

Add video decoder basic config to Kconfig and
select the required V4l2 modules.
Add video decoder to Makefile

Signed-off-by: Angela Stegmaier <angelabaker@ti.com>
Signed-off-by: Sunita Nadampalli <sunitan@ti.com>
Signed-off-by: Sidraya Jayagond <sidraya.bj@pathparntertech.com>
---
 drivers/media/platform/Kconfig  | 12 ++++++++++++
 drivers/media/platform/Makefile |  2 ++
 2 files changed, 14 insertions(+)

diff --git a/drivers/media/platform/Kconfig b/drivers/media/platform/Kconfig
index e200503f72e4..5a5fc3aa1dea 100644
--- a/drivers/media/platform/Kconfig
+++ b/drivers/media/platform/Kconfig
@@ -507,6 +507,18 @@ config VIDEO_TI_VPE_DEBUG
 	help
 	  Enable debug messages on VPE driver.
 
+config VIDEO_IMG_VXD_DEC
+	  tristate "IMG VXD DEC (Video Decoder) driver"
+	  depends on VIDEO_DEV && VIDEO_V4L2
+	  select VIDEOBUF2_CORE
+	  select VIDEOBUF2_DMA_CONTIG
+	  select VIDEOBUF2_DMA_SG
+	  select V4L2_MEM2MEM_DEV
+	  help
+	    This is an IMG VXD DEC V4L2 driver that adds support for the
+	    Imagination D5520 (Video Decoder) hardware.
+	    The module name when built is vxd-dec.
+
 config VIDEO_QCOM_VENUS
 	tristate "Qualcomm Venus V4L2 encoder/decoder driver"
 	depends on VIDEO_DEV && VIDEO_V4L2
diff --git a/drivers/media/platform/Makefile b/drivers/media/platform/Makefile
index 8a7fd3a37c56..f912e273c9d5 100644
--- a/drivers/media/platform/Makefile
+++ b/drivers/media/platform/Makefile
@@ -80,3 +80,5 @@ obj-$(CONFIG_VIDEO_QCOM_CAMSS)		+= qcom/camss/
 obj-$(CONFIG_VIDEO_QCOM_VENUS)		+= qcom/venus/
 
 obj-y					+= sunxi/
+
+obj-$(CONFIG_VIDEO_IMG_VXD_DEC)		+= vxe-vxd/decoder/
-- 
2.17.1

